packages/celotool/src/cmds/account/faucet.ts

```diff
@@ -76,45 +98,49 @@ export const handler = async (argv: FaucetArgv) => {
     console.log(`Using account: ${account}`)
     kit.connection.defaultAccount = account
 
-    const [goldToken, stableToken, reserve] = await Promise.all([
-      kit.contracts.getGoldToken(),
-      kit.contracts.getStableToken(),
-      kit.contracts.getReserve(),
-    ])
+    const faucetToken = async (tokenParams: TokenParams) => {
+      if (!tokenParams.amount) {
+        return
+      }
 
-    if (argv.checkzero) {
+      const tokenWrapper = await kit.celoTokens.getWrapper(tokenParams.token as any)
       for (const address of addresses) {
-        // Check this address hasn't already been fauceted and has zero gold
-        if (
-          (argv.gold !== 0 && !(await goldToken.balanceOf(address)).isZero()) ||
-          (argv.dollar !== 0 && !(await stableToken.balanceOf(address)).isZero())
-        ) {
-          console.error(
-            `Unable to faucet ${address} on ${argv.celoEnv}: --checkzero specified, but balance is non-zero`
-          )
-          process.exit(1)
+        if (argv.checkzero) {
+          // Check if address account balance of this token is zero
```

Comment:
nit: "is not zero"

Reply:
Just to point a detail
Here we changed the behaviour (I'm ok with this, but pointing out just in case it depends on something like a e2e test or something)
But the old implementation failed if one of the balances was zero, in this case, as is called for every token, one of the calls could succeed because for that specific token, it has no balance.

Reply:
yes a good point -- my sense is that this was a historic holdover to prevent double-fauceting the same address (+ not really used anymore, also no uses in tests that I could find) so will leave as is for now!